drivers/overlay/ostweaks_linux.go

```diff
@@ -21,10 +22,43 @@ func writeSystemProperty(key, value string) error {
 	return ioutil.WriteFile(path.Join("/proc/sys", keyPath), []byte(value), 0644)
 }
 
+func readSystemProperty(key string) ([]byte, error) {
+	keyPath := strings.Replace(key, ".", "/", -1)
+	return ioutil.ReadFile(path.Join("/proc/sys", keyPath))
+}
+
+func byte2int(b []byte) int {
+	s := string(b)
+	s2 := s[:len(s)-1] // remove trailing newline
+	i := str2int(s2)
+	return i
+}
+
+func str2int(s string) int {
+	i, err := strconv.Atoi(s)
+	if err != nil {
+		logrus.Errorf("error converting string %v to int, err: %s", s, err)
+	}
+	return i
+}
+
 func applyOStweaks() {
 	for k, v := range sysctlConf {
-		if err := writeSystemProperty(k, v); err != nil {
-			logrus.Errorf("error setting the kernel parameter %s = %s, err: %s", k, v, err)
+		oldv_b, err := readSystemProperty(k)
+
+		if err != nil {
+			logrus.Errorf("error reading the kernel parameter %s = %s, err: %s", k, v, err)
+		}
+
+		oldv_i := byte2int(oldv_b)
```

Comment:
error management does not seem to be correct. If the readSystem fails, you just write an error log but then you try to parse the value anyway.